src/brpc/policy/redis_protocol.cpp

```diff
@@ -113,6 +140,69 @@ int ConsumeCommand(RedisConnContext* ctx,
             }
         }
     }
+    else if (args[0] == "scan") {
+        // find the cursor param
+        if (args.size() < 2) {
+            output.SetError("ERR wrong number of arguments for 'scan' command");
+            output.SerializeTo(appender);
+            return 0;
+        }
+        auto cursor_uint64 = args[1];
+        auto iterator = ctx->scan_cursors.find(cursor_uint64.as_string());
+        std::vector<butil::StringPiece> mut_args;
+        for (auto& arg : args) {
+            mut_args.push_back(arg);
+        }
+        if (cursor_uint64 == "0") {
+        } else if (iterator != ctx->scan_cursors.end()) {
+            // cursor found, use the value
+            mut_args[1] = iterator->second;
+        } else {
+            // cursor not found, return corrutped cursor error
```

Comment:
It seems scan cursor can be any random integer. If the cursor not found, just scan normally.